execute/plugin.go

```diff
@@ -151,25 +144,30 @@ func getPendingExecutedReports(
 			continue
 		}
 
+		sort.Slice(reports, func(i, j int) bool {
+			return reports[i].SequenceNumberRange.Start() < reports[j].SequenceNumberRange.Start()
+		})
```

Comment:
Not 100% sure but isn't this kind of sort already being done by the `CommitReportsGTETimestamp` method? Generally, performance wise, I think its better to do those kinds of ops as part of the DB query especially if the query needs to use a sorted index anyway.

If its _not_ being done there I think we should create a ticket to do it there + have the sorted order enforced via tests + godoc on the method.

cc @0xAustinWang @mateusz-sekara